src/zenml/pipelines/base_pipeline.py

```diff
@@ -499,6 +502,42 @@ def run(
                     update={"pipeline_id": pipeline_id}
                 )
 
+            if schedule:
```

Comment:
Not a huge fan of registering the schedule here without any checks whether it's possible (and if it will even get until the orchestrator, there are lots of failure points even before getting there). I'd say definitely add a comment here that this is to be changed in the future if reworking it and making it nicer doesn't work for now

Reply:
Yeah me neither, but I would keep it for now because it's actually quite useful for testing. Added a "TODO" nonetheless.